src/bp/ui-shared/src/FileDisplay/index.tsx

```diff
@@ -0,0 +1,51 @@
+import { Button, Intent, Position, Tooltip } from '@blueprintjs/core'
+import mime from 'mime-types'
+import React, { FC } from 'react'
+import { lang } from '../translations'
+import style from './style.scss'
+import { FileDisplayProps } from './typings'
+
+const FileDisplay: FC<FileDisplayProps> = props => {
+  const { url, type, deletable, onDelete } = props
+
+  const mimeType = mime.lookup(url) || undefined
+
+  const deletableFile = () => (
+    <Tooltip content={lang('delete')} position={Position.TOP}>
+      <Button className={style.deleteFile} minimal small intent={Intent.DANGER} icon="trash" onClick={onDelete} />
+    </Tooltip>
+  )
+
+  switch (type) {
+    case 'image':
+      return (
+        <div style={{ backgroundImage: `url('${url}')` }} className={style.imageWrapper}>
+          <div className={style.imageWrapperActions}>{deletable && deletableFile()}</div>
+        </div>
+      )
+    case 'audio':
+      return (
+        <div className={style.audioWrapper}>
+          <div className={style.audioWrapperActions}>{deletable && deletableFile()}</div>
+          <audio controls className={style.audioWrapperSource}>
+            <source src={url} type={mimeType} />
+            Your browser does not support the audio element.
```

Comment:
this is taken out of the dom ? if the browser supports it ?

Reply:
Not sure I understand your question here! 

Here's what documentation I used when adding this feature: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/audio

Reply:
Hmm what I meant is :

Is this message taken out of the DOM if the browser supports it ? But now I get it. thanks